backend/dataall/modules/shares_base/services/share_item_service.py

```diff
@@ -127,12 +128,14 @@ def add_shared_item(uri: str, data: dict = None):
             share_sm = ShareObjectSM(share.status)
             new_share_state = share_sm.run_transition(ShareItemActions.AddItem.value)
             share_sm.update_state(session, share, new_share_state)
```

Comment:
`get_share_item` queried different db tables depending on the itemType:

```
    @staticmethod
    def get_share_item(session, item_type, item_uri):
        if item_type == ShareableType.Table.value:
            return session.query(DatasetTable).get(item_uri)
        if item_type == ShareableType.StorageLocation.value:
            return session.query(DatasetStorageLocation).get(item_uri)
        if item_type == ShareableType.S3Bucket.value:
            return session.query(DatasetBucket).get(item_uri)
```
Instead, now we get the processor that corresponds to the itemType and query the processor.shareableType